packages/playground/optimize-deps/package.json

```diff
@@ -15,6 +15,8 @@
     "dep-linked": "link:./dep-linked",
     "dep-linked-include": "link:./dep-linked-include",
     "dep-esbuild-plugin-transform": "link:./dep-esbuild-plugin-transform",
+    "dep-cjs-compiled-from-esm": "file:./dep-cjs-compiled-from-esm",
```

Comment:
It really confusing me that if I use `link:` instead of `file:` Vite won't pre-bundle my package. Not sure whether it is a bug too?

Reply:
Maybe vite gets a real address with no `node_modules` in the path, so it's not pre-bundle
https://github.com/vitejs/vite/blob/eef51cb37db98a1ad9a541bdd3cd74736ff8488d/packages/vite/src/node/optimizer/scan.ts#L298-L302

this [issue comment](https://github.com/vitejs/vite/issues/4556#issuecomment-898862114) analyzes the issue

Reply:
Yes. It looks like `file:` will make npm/yarn copy the package to node_modules rather than link it.